File: apps/uk/replace.js
const fs = require('fs');
const path = require('path');

const files = [
  'src/app/page.tsx',
  'src/app/not-found.tsx',
  'src/app/layout.tsx',
  'src/data/about.ts',
  'src/data/contact.ts',
  'src/data/faqs.ts',
  'src/lib/site.ts',
  'src/lib/seo.ts',
  'src/lib/market.ts'
];

const basePath = 'e:\\1st YEAR DTU\\New folder\\muuhu\\apps\\uk';

function replaceAll(text) {
  let t = text;
  t = t.replace(/Buudy LED Mask/g, 'Muuhu IPL');
  t = t.replace(/buudy-led-mask/g, 'muuhu-ipl-hair-removal');
  t = t.replace(/LED Face Mask/g, 'IPL Hair Removal');
  t = t.replace(/LED face mask/g, 'IPL hair removal');
  t = t.replace(/LED Mask/g, 'IPL');
  t = t.replace(/Buudy/g, 'Muuhu');
  t = t.replace(/buudy/g, 'muuhu');
  return t;
}

function processFile(filePath) {
  const fullPath = path.join(basePath, filePath);
  if (!fs.existsSync(fullPath)) {
    console.log(`File not found: ${fullPath}`);
    return;
  }
  
  const content = fs.readFileSync(fullPath, 'utf-8');
  const pattern = /(\/images\/[^"'\s]+|\/media\/[^"'\s]+)/g;
  
  const segments = content.split(pattern);
  let newContent = '';
  
  for (let i = 0; i < segments.length; i++) {
    if (i % 2 === 1) {
      // Protected path
      newContent += segments[i];
    } else {
      newContent += replaceAll(segments[i]);
    }
  }
  
  fs.writeFileSync(fullPath, newContent, 'utf-8');
  console.log(`Processed: ${filePath}`);
}

files.forEach(processFile);
console.log('Done replacing.');

File: apps/uk/replace.py
import re
import sys
import os

def process_file(path):
    with open(path, 'r', encoding='utf-8') as f:
        text = f.read()
    
    def replace_all(t):
        t = t.replace('Buudy LED Mask', 'Muuhu IPL')
        t = t.replace('buudy-led-mask', 'muuhu-ipl-hair-removal')
        t = t.replace('LED Face Mask', 'IPL Hair Removal')
        t = t.replace('LED face mask', 'IPL hair removal')
        t = t.replace('LED Mask', 'IPL')
        t = t.replace('Buudy', 'Muuhu')
        t = t.replace('buudy', 'muuhu')
        return t
        
    new_text = ''
    
    pattern = r'(/images/[^\"\'\s]+|/media/[^\"\'\s]+)'
    segments = re.split(pattern, text)
    
    for i, segment in enumerate(segments):
        if i % 2 == 1:
            # This is the protected path
            new_text += segment
        else:
            new_text += replace_all(segment)
            
    with open(path, 'w', encoding='utf-8') as f:
        f.write(new_text)

files = [
    r'src\app\page.tsx',
    r'src\app\not-found.tsx',
    r'src\app\layout.tsx',
    r'src\data\about.ts',
    r'src\data\contact.ts',
    r'src\data\faqs.ts',
    r'src\lib\site.ts',
    r'src\lib\seo.ts',
    r'src\lib\market.ts'
]

for f in files:
    full_path = os.path.join(r'e:\1st YEAR DTU\New folder\muuhu\apps\uk', f)
    if os.path.exists(full_path):
        process_file(full_path)
    else:
        print(f"File not found: {full_path}")

print('Done replacing.')
